### ISSUE TITLE
Julia 0.5.0 from Generic Linux binaries fails on startup with libgit2 error

### ISSUE THREAD

**Issue by mgerber79** [JuliaLang/julia#18753]
Hi all, I'm running Gentoo Linux x86_64. I installed the Julia v0.5.0 Generic Linux binary 64-bit (X86). Calling Julia gives the following error:

$ julia
fatal: error thrown and no exception handler available.
Base.InitError(mod=:LibGit2, error=ErrorException("could not load library "libgit2"
libgssapi_krb5.so.2: cannot open shared object file: No such file or directory"))
rec_backtrace at /home/centos/buildbot/slave/package_tarball64/build/src/stackwalk.c:84
record_backtrace at /home/centos/buildbot/slave/package_tarball64/build/src/task.c:232
jl_throw at /home/centos/buildbot/slave/package_tarball64/build/src/task.c:550
jl_errorf at /home/centos/buildbot/slave/package_tarball64/build/src/builtins.c:78
jl_dlerror at /home/centos/buildbot/slave/package_tarball64/build/src/dlload.c:69 [inlined]
jl_load_dynamic_library_ at /home/centos/buildbot/slave/package_tarball64/build/src/dlload.c:209
jl_get_library at /home/centos/buildbot/slave/package_tarball64/build/src/runtime_ccall.cpp:152
jl_load_and_lookup at /home/centos/buildbot/slave/package_tarball64/build/src/runtime_ccall.cpp:163
unknown function (ip: 0x7f4c5fed6346)
__init__ at ./libgit2/libgit2.jl:538
unknown function (ip: 0x7f4c5fed6b58)
jl_call_method_internal at /home/centos/buildbot/slave/package_tarball64/build/src/julia_internal.h:189 [inlined]
jl_apply_generic at /home/centos/buildbot/slave/package_tarball64/build/src/gf.c:1942
jl_apply at /home/centos/buildbot/slave/package_tarball64/build/src/julia.h:1392 [inlined]
jl_module_run_initializer at /home/centos/buildbot/slave/package_tarball64/build/src/toplevel.c:83
_julia_init at /home/centos/buildbot/slave/package_tarball64/build/src/init.c:742
julia_init at /home/centos/buildbot/slave/package_tarball64/build/src/task.c:283
unknown function (ip: 0x4013aa)
__libc_start_main at /lib64/libc.so.6 (unknown line)
unknown function (ip: 0x4013fc)

Does anybody have an idea?

**Comment by mauro3** [comment]
X-ref to same(?) problem: https://groups.google.com/d/msg/julia-users/Xd3MdrK4ucU/CdlOR3pLCAAJ

**Comment by felipenoris** [comment]
try emerge mit-krb5

**Comment by tkelman** [comment]
duplicate of https://github.com/JuliaLang/julia/issues/18693

**Comment by tkelman** [comment]
Sorry, actually not a duplicate of #18693, but was fixed by https://github.com/JuliaLang/julia/issues/18658. We just haven't had new nightlies including that fix because of #18693.

**Comment by marius311** [comment]
Hi, I'm getting this error on one of my systems as well. Is there anything I can do, like upgrading Git on this system or something like that, that would allow me to use Julia 0.5 rather than compiling a more recent version myself or using the nightlies?

**Comment by tkelman** [comment]
Install krb5.

**Comment by marius311** [comment]
Thanks for the quick reply. Just installed krb5 and still getting the error. Should I be seeing libkrb5.so being loaded by the julia executable or anything like that? I don't see any mention of krb if I `strace julia`.

**Comment by tkelman** [comment]
What distribution are you using?

**Comment by marius311** [comment]
This is a fairly ancient SUSE Linux Enterprise Server 10 SP4. The krb5 libraries I compiled / installed the latest version myself in my user directory (as I don't have admin privileges on this cluster), but like mentioned above, I don't think the executable ever gets to loading them as I don't see them with strace.

**Comment by tkelman** [comment]
So a distro version that has been unsupported for 3 years? Try adding to your LD_LIBRARY_PATH

**Comment by marius311** [comment]
> So a distro version that has been unsupported for 3 years? 

Correct :) so I cerainly won't be upset if I can't get help with this. But this cluster has 100 underused nodes if I can make it work... 

Anyway, yes, $HOME/lib contains libgssapi_krb5.so*, lib/libkrb5.so*, and libkrb5support.so*, and is in $LD_LIBRARY_PATH.

**Comment by tkelman** [comment]
Check ldd on libgit2.so and libssl.so in the Julia binaries to see what else you might be missing.

**Comment by marius311** [comment]
Ah, apologies, the error message actually changed after your initial suggestion of installing krb5 but I didn't notice, it now reads `Base.InitError(mod=:LibGit2, error=DivideError()))`. I also checked with `ldd` and `libssl.so` was missing `libcrypto.so`. In any case, this all made me think the path of least resistance might instead be to try and compile Julia on this system directly. I went ahead with that and after a few hiccups it seems to be working. Thanks for the help!
